scipy/integrate/_ivp/lsoda.py

```diff
@@ -0,0 +1,190 @@
+        approximated by finite differences. It is generally recommended to
+        provide the Jacobian rather than relying on a finite difference
+        approximation.
+    lband, uband : int or None, optional
+        Jacobian band width:
+        ``jac[i, j] != 0 only for i - lband <= j <= i + uband``. Setting these
+        requires your jac routine to return the Jacobian in the packed format:
+        the returned array must have ``n`` columns and ``uband + lband + 1``
+        rows in which Jacobian diagonals are written. Specifically
+        ``jac_packed[uband + i - j , j] = jac[i, j]``. The same format is used
+        in `scipy.linalg.solve_banded` (check for an illustration).
+        These parameters can be also used with ``jac=None`` to reduce the
+        number of Jacobian elements estimated by finite differences.
+    vectorized : bool, optional
+        Whether `fun` is implemented in a vectorized fashion. A vectorized
+        implementation offers no advantages for this solver. Default is False.
+
+    Attributes
+    ----------
+    n : int
+        Number of equations.
+    status : string
+        Current status of the solver: 'running', 'finished' or 'failed'.
+    t_crit : float
+        Boundary time.
+    direction : float
+        Integration direction: +1 or -1.
+    t : float
+        Current time.
+    y : ndarray
+        Current state.
+    t_old : float
+        Previous time. None if no steps were made yet.
+    nfev : int
+        Number of the system's rhs evaluations.
+    njev : int
+        Number of the Jacobian evaluations.
+
+    References
+    ----------
+    .. [1] A. C. Hindmarsh, "ODEPACK, A Systematized Collection of ODE
+           Solvers," IMACS Transactions on Scientific Computation, Vol 1.,
+           pp. 55-64, 1983.
+    .. [2] L. Petzold, "Automatic selection of methods for solving stiff and
+           nonstiff systems of ordinary differential equations", SIAM Journal
+           on Scientific and Statistical Computing, Vol. 4, No. 1, pp. 136-148,
+           1983.
+    """
+    def __init__(self, fun, t0, y0, t_crit, first_step=None, min_step=0.0,
+                 max_step=np.inf, rtol=1e-3, atol=1e-6, jac=None, lband=None,
+                 uband=None, vectorized=False, **extraneous):
+        warn_extraneous(extraneous)
+        super(LSODA, self).__init__(fun, t0, y0, t_crit, vectorized)
+
+        if first_step is None:
+            first_step = 0  # LSODA value for automatic selection.
+        elif first_step <= 0:
+            raise ValueError("`first_step` must be positive or None.")
+
+        if max_step == np.inf:
+            max_step = 0  # LSODA value for infinity.
+        elif max_step <= 0:
+            raise ValueError("`max_step` must be positive.")
+
+        if min_step < 0:
+            raise ValueError("`min_step` must be nonnegative.")
+
+        rtol, atol = validate_tol(rtol, atol, self.n)
+
+        if jac is None:  # No lambda as PEP8 insists.
```

Comment:
@drhagen I changed that thing, it works fine and more clear now.

Also I took a liberty to remove all LSODA attributes from this class entirely. Is it OK decision you think?